tiny_gltf.h

```diff
@@ -4442,8 +4444,17 @@ static bool ParseImage(Image *image, const int image_idx, std::string *err,
     return false;
   }
 
-  return LoadImageData(image, image_idx, err, warn, 0, 0, &img.at(0),
-                       static_cast<int>(img.size()), load_image_user_data);
+  bool ok = LoadImageData(image, image_idx, err, warn, 0, 0, &img.at(0),
+                          static_cast<int>(img.size()), load_image_user_data);
+
+  // If the image is encoded as data URI, and the actual image data was
+  // not extracted, then we keep the original data URI that holds all
+  // the data.
+  if (IsDataURI(uri) && ok && image->image.empty()) {
```

Comment:
Keep the uri only if the data has not been loaded. Otherwise we have it in memory twice.